### PR TITLE
Added pubsubConfig and webhookConfig support to the cloud build resource.

### PR DESCRIPTION
**Pull request by iamsumit**
Google have [API documentation](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.triggers) to create a pub/sub and webhook trigger in the cloud build resource, but, [the terraform google provider](https://github.com/hashicorp/terraform-provider-google/issues/8692) [doesn't](https://github.com/hashicorp/terraform-provider-google/issues/9189). I have updated the terraform resource to add that support.

Fixes https://github.com/hashicorp/terraform-provider-google/issues/8692, https://github.com/hashicorp/terraform-provider-google/issues/9189

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudbuild: Added `pubsub_config` and `webhook_config` parameter to `google_cloudbuild_trigger`.
```

### CODE REVIEW

**Comment by iamsumit** [comment]
@googlebot I signed it!

**Comment by iamsumit** [comment]
@googlebot I fixed it.

**Comment by c2thorn** [comment]
@iamsumit, sorry about this, but could you rebase off of the latest master commit? We changed our CI and unfortunately it means our tests won't run without doing so.

**Comment by iamsumit** [comment]
@c2thorn I have rebased it. Thank you.

**Comment by iamsumit** [comment]
@c2thorn Is there anything I have to do for the failing test?

File: mmv1/products/cloudbuild/api.yaml [hunk 1]
```diff
@@ -230,6 +237,54 @@ objects:
...
+          a build whenever a Pub/Sub message is published.
+        exactly_one_of:
+          - trigger_template
+          - github
+          - pubsub_config
+          - webhook_config
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'subscription'
+            description: |
```
Comment (c2thorn) [inline mmv1/products/cloudbuild/api.yaml:253]: Fields that are output only should have the property
```
output: true
```

File: mmv1/products/cloudbuild/api.yaml [hunk 2]
```diff
@@ -230,6 +237,54 @@ objects:
...
+              Output only. Name of the subscription.
+          - !ruby/object:Api::Type::String
+            name: 'topic'
```
Comment (c2thorn) [inline mmv1/products/cloudbuild/api.yaml:256]: I am assuming that if `pubsubConfig` is defined in the API, `topic` is a necessary field. We can require this in the Terraform schema by adding
```
required: true
```
and should do so for any other necessary fields.

File: mmv1/products/cloudbuild/api.yaml [hunk 3]
```diff
@@ -230,6 +237,54 @@ objects:
...
+            description: |
+              The name of the topic from which this subscription is receiving messages.
+          - !ruby/object:Api::Type::String
+            name: 'service_account_email'
+            description: |
+              Service account that will make the push request.
+          - !ruby/object:Api::Type::String
+            name: 'state'
```
Comment (c2thorn) [inline mmv1/products/cloudbuild/api.yaml:264]: Looks like it should have `output: true`

File: mmv1/products/cloudbuild/api.yaml [hunk 4]
```diff
@@ -230,6 +237,54 @@ objects:
...
+          - github
+          - pubsub_config
+          - webhook_config
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'secret'
+            description: |
+              Resource name for the secret required as a URL parameter.
+          - !ruby/object:Api::Type::String
+            name: 'state'
```
Comment (c2thorn) [inline mmv1/products/cloudbuild/api.yaml:284]: `output: true` here as well

File: mmv1/products/cloudbuild/api.yaml [hunk 5]
```diff
@@ -230,6 +237,54 @@ objects:
...
+        exactly_one_of:
+          - trigger_template
+          - github
+          - pubsub_config
+          - webhook_config
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'secret'
+            description: |
+              Resource name for the secret required as a URL parameter.
```
Comment (c2thorn) [inline mmv1/products/cloudbuild/api.yaml:282]: Let's add `required: true` here

**Review by c2thorn — CHANGES_REQUESTED** [review]
Looks solid, just requesting a few schema changes. Thanks for adding the update tests in there as well!

File: mmv1/products/cloudbuild/api.yaml (hunk 1 above)
Comment (iamsumit) [reply mmv1/products/cloudbuild/api.yaml:253]: My bad. I missed It here. Thank you. I have updated it now.

File: mmv1/products/cloudbuild/api.yaml (hunk 2 above)
Comment (iamsumit) [reply mmv1/products/cloudbuild/api.yaml:257]: Updated.

File: mmv1/products/cloudbuild/api.yaml (hunk 3 above)
Comment (iamsumit) [reply mmv1/products/cloudbuild/api.yaml:266]: Updated.

File: mmv1/products/cloudbuild/api.yaml (hunk 4 above)
Comment (iamsumit) [reply mmv1/products/cloudbuild/api.yaml:288]: Updated.

File: mmv1/products/cloudbuild/api.yaml (hunk 5 above)
Comment (iamsumit) [reply mmv1/products/cloudbuild/api.yaml:286]: Updated.

**Comment by iamsumit** [comment]
@c2thorn A gentle reminder to review this PR. Please do let me know if there is anything I am missing?

**Review by c2thorn — APPROVED** [review]
Relevant tests have passed

**Comment by iamsumit** [comment]
@c2thorn So, I checked that the 3.75 includes the changes but when I used that on my local, it didn't work. Did we miss anything?

https://github.com/hashicorp/terraform-provider-google/compare/v3.75.0...master

**Comment by c2thorn** [comment]
@iamsumit the release was cut before this change. https://github.com/hashicorp/terraform-provider-google/commits/release-3.75.0 shows the commits in 3.75, I believe your link shows the differences between `v3.75.0` and `master`

**Comment by iamsumit** [comment]
@c2thorn Aah. My bad. Thank you for confirming.

**Comment by ffjeremy** [comment]
I've updated to version v3.77.0, but it appears that this feature is still missing some functionality. I do not see a way to specify the repo/branch info. These fields are required when creating a pubsub trigger from the GCP console. Did I miss something in the docs, or is feature not complete yet?

**Comment by iamsumit** [comment]
> I've updated to version v3.77.0, but it appears that this feature is still missing some functionality. I do not see a way to specify the repo/branch info. These fields are required when creating a pubsub trigger from the GCP console. Did I miss something in the docs, or is feature not complete yet?

There is no parameter in [Google API documentation ](https://cloud.google.com/build/docs/api/reference/rest/v1/projects.locations.triggers#BuildTrigger.PubsubConfig) to specify this information. I know it is required in the UI, but it seems like a bug, and there is no need to specify those in the terraform or over the rest API.

**Comment by ffjeremy** [comment]
@iamsumit It seems that `pubsubConfig` should not be mutually exclusive with `github|triggerTemplate`. Perhaps that is the missing piece. Otherwise there is no way to actually run any code.

**Comment by c2thorn** [comment]
I would suggest submitting a new Github issue outlining the problem here so it can get assigned and tracked.

**Comment by tuananhnguyen-ct** [comment]
I have the same issue using webhookConfig being unable to specify a source repo

**Comment by bbhoss** [comment]
I guess this currently only allows you to trigger cloudbuilds defined inline without any associated repository? It seems like at the very least you could allow `trigger_template` with `pubsub_config` or `webhook_config`. That way someone could still manually create a cloud source repo and enable the integration/sync using the GUI. Not ideal at all considering that via the GCP UI you can create pubsub and webhook or even manually triggered builds using the github app without selecting push/tag/etc trigger. But it would be better than totally blocking with no workarounds.

**Pull request merged**